components/game/screen.js

```diff
@@ -39,94 +43,191 @@ class GameScreen {
+        borderWidth * y + blockSize * y + topMargin,
+        blockSize,
+        blockSize
+    );
+  }
 
-        this.ctx.fillRect(
-          borderWidth * (x - outBorderBlockCount) +
-            blockSize * (x - outBorderBlockCount) +
-            leftMargin,
-          borderWidth * y + blockSize * y + topMargin,
-          blockSize,
-          blockSize
-        );
 
-        //draw top left shadow
-        this.ctx.fillStyle = blockColor + blockShadowOpacity2;
-        this.ctx.fillRect(
+  drawTopLeftShadow(x, y, ctx, color) {
+    ctx.fillStyle = color + blockShadowOpacity2;
+    ctx.fillRect(
+        borderWidth * (x - outBorderBlockCount) +
+        blockSize * (x - outBorderBlockCount) +
+        leftMargin,
+        borderWidth * y + blockSize * y + topMargin,
+        blockSize,
+        shadowWidth
+    );
+    ctx.fillRect(
+        borderWidth * (x - outBorderBlockCount) +
+        blockSize * (x - outBorderBlockCount) +
+        leftMargin,
+        borderWidth * y + blockSize * y + topMargin,
+        shadowWidth,
+        blockSize
+    );
+  }
+
+  drawBottomRightShadow(x, y, ctx, color) {
+      ctx.fillStyle = color;
+      ctx.fillRect(
           borderWidth * (x - outBorderBlockCount) +
-            blockSize * (x - outBorderBlockCount) +
-            leftMargin,
-          borderWidth * y + blockSize * y + topMargin,
+          blockSize * (x - outBorderBlockCount) +
+          leftMargin,
+          borderWidth * y +
+          blockSize * y +
+          topMargin +
+          blockSize -
+          shadowWidth,
           blockSize,
           shadowWidth
-        );
-
-        this.ctx.fillRect(
+      )
+      ctx.fillRect(
           borderWidth * (x - outBorderBlockCount) +
-            blockSize * (x - outBorderBlockCount) +
-            leftMargin,
+          blockSize * (x - outBorderBlockCount) +
+          leftMargin +
+          blockSize -
+          shadowWidth,
           borderWidth * y + blockSize * y + topMargin,
           shadowWidth,
           blockSize
-        );
-
-        //draw bottom right shadow
-        if (controlSingleBlock.isExist || stackedSingleBlock.isExist) {
-          this.ctx.fillStyle = blockColor;
-          this.ctx.fillRect(
-            borderWidth * (x - outBorderBlockCount) +
-              blockSize * (x - outBorderBlockCount) +
-              leftMargin,
-            borderWidth * y +
-              blockSize * y +
-              topMargin +
-              blockSize -
-              shadowWidth,
-            blockSize,
-            shadowWidth
-          );
-
-          this.ctx.fillRect(
-            borderWidth * (x - outBorderBlockCount) +
-              blockSize * (x - outBorderBlockCount) +
-              leftMargin +
-              blockSize -
-              shadowWidth,
-            borderWidth * y + blockSize * y + topMargin,
-            shadowWidth,
-            blockSize
-          );
-        }
-      }
+      );
+  }
+
+  clearPrevRect(x, y, ctx) {
```

Comment:
This is necessary to wipe previously rendered block. It might be need refactoring.